emulators/SyncEmulator.py

```diff
@@ -106,8 +108,10 @@ def print_statistics(self):
 
     def terminated(self, index:int):
         self._progress.acquire()
+        self._done[index] = True
         self._terminated += 1
-        if self._terminated >= len(self._devices):
+        if all([self._done[x] or not self._threads[x].is_alive()
+                for x in self.ids()]):
             if self._round_lock.locked():
```

Comment:
@anton-christensen is there some indentation that has gone wrong here?

Reply:
Small stylistic choice, but its a short line so i've moved it onto the same line.